staging/src/k8s.io/apiserver/pkg/storage/etcd3/latency_tracker.go

```diff
@@ -21,14 +21,15 @@ import (
 	"time"
 
 	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/client/v3/kubernetes"
 	endpointsrequest "k8s.io/apiserver/pkg/endpoints/request"
 )
 
 // NewETCDLatencyTracker returns an implementation of
 // clientv3.KV that times the calls from the specified
 // 'delegate' KV instance in order to track latency incurred.
-func NewETCDLatencyTracker(delegate clientv3.KV) clientv3.KV {
-	return &clientV3KVLatencyTracker{KV: delegate}
+func NewETCDLatencyTracker(delegate *kubernetes.Client) (clientv3.KV, kubernetes.Interface) {
+	return &clientV3KVLatencyTracker{KV: delegate.Client.KV}, &kubernetesInterfaceLatencyTracker{Interface: delegate.Kubernetes}
```

Comment:
`kubernetes.Client` is guaranteed to implement the `kubernetes.Interface`.
```suggestion
	return &clientV3KVLatencyTracker{KV: delegate.Client.KV}, &kubernetesInterfaceLatencyTracker{Interface: delegate}
```

Reply:
If you follow the suggestion https://github.com/etcd-io/etcd/pull/16333#discussion_r1669149886, then this can be as below,

```
return &clientV3KVLatencyTracker{KV: delegate}, &kubernetesInterfaceLatencyTracker{Interface: delegate}
```

Reply:
This will not work, it causes infinite recursion.